app/scripts/modules/core/src/application/nav/NavCategory.tsx

```diff
@@ -1,19 +1,20 @@
 import React from 'react';
 
 import { DataSourceNotifications } from 'core/entityTag/notifications/DataSourceNotifications';
-import { Icon, useDataSource } from '../../presentation';
+import { Icon, Tooltip, useDataSource } from '../../presentation';
 
 import { ApplicationDataSource } from '../service/applicationDataSource';
 import { Application } from '../application.model';
 import { IEntityTags } from '../../domain';
 
 export interface INavCategoryProps {
+  app: Application;
   category: ApplicationDataSource;
   isActive: boolean;
-  app: Application;
+  isCollapsed: boolean;
 }
 
-export const NavCategory = ({ app, category, isActive }: INavCategoryProps) => {
+export const NavCategory = ({ app, category, isActive, isCollapsed }: INavCategoryProps) => {
```

Comment:
This component is pretty confusing to me.  Is it a Category?  A link/nav item? A data source?  Is a category a collection of items? What is the relationship between the datasource prop and the other datasource inside the component?   

It seems like this might be easier to grok if we separated out the data flow from and extract a purely presentational component, something like:

```js
<DataSourceThing 
  dataSource={datasource} 
  otherDataRelatedProp={otherDataRelatedProp}
  render={navItemMarkupProps => <NavItemMarkupOnly {...navItemMarkupProps} />} 
/>
<PagerDutyThing render={navItemMarkupProps => <NavItemMarkupOnly {...navItemMarkupProps} />} />
``` 
obviously with better names

Reply:
funny, I just noticed that @vigneshm had almost the exact same feedback

Reply:
I can change the name to `NavItem`, I was trying to stay consistent with legacy naming conventions, but if it confusing I am happy to change it.  Historically, the category was equivalent to a datasource.

I think it is too much abstraction to separate into 2 components. All this one is doing is subscribing to the data source, and then rendering the item with the running count, icon, name and alerts.